changelog/@unreleased/pr-38.v2.yml

```
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Make `IdeaConfigurationPlugin` configuration-cache friendly
```

```suggestion
  description: Make `IdeaConfigurationPlugin` configuration-cache compatible.
```

Assuming you're running this on the `classes` task.

Is there a test to make sure we don't regress?

---

Hmm, I think this PR makes IdeaConfigurationPlugin compatible with configuration cache in general.

---

I've updated the suggestion to just be `compatible`.

Question about test still remains

---

Apologies I missed that. And yep, I've added a test that (I think) covers the entire plugin.